src/ParameterLists.f90

```diff
@@ -3062,7 +3127,319 @@ FUNCTION match_ParamType(thisParam,thatParam,prefix) RESULT(bool)
         CLASS DEFAULT
           CONTINUE
       ENDSELECT
-    ENDFUNCTION match_ParamType
+    ENDFUNCTION matchTest_ParamType
+!
+!-------------------------------------------------------------------------------
+!> @brief This function assumes that thisParam and thatParam are of the same
```

Comment:
Split this into a more brief description and then `Implementation Notes` where you should list assumptions about the routine.

Reply:
So this one checks list structure and values? Not just structure?

Reply:
I'll make those mods.  Yes, structure and values are checked.

Reply:
Updated.